scenarios/ebb_and_flow.go

```diff
@@ -0,0 +1,251 @@
+			}).run(ctx)
+		}),
+	})
+}
+
+type ebbAndFlow struct {
+	loadgen.ScenarioInfo
+	rng *rand.Rand
+
+	startTime      time.Time
+	generatedCount atomic.Int64
+	processedCount atomic.Int64
+}
+
+func (e *ebbAndFlow) run(ctx context.Context) error {
+	e.startTime = time.Now()
+
+	// Parse and validate scenario options.
+	minBacklog := int64(e.ScenarioOptionInt("min-backlog", 10))
+	maxBacklog := int64(e.ScenarioOptionInt("max-backlog", 30))
+	frequency := e.ScenarioOptionDuration("frequency", 60*time.Second)
+	sleepDuration := e.ScenarioOptionDuration("sleep-duration", 1*time.Millisecond)
+	maxRate := e.ScenarioOptionInt("max-rate", 1000)
+	controlInterval := e.ScenarioOptionDuration("control-interval", 100*time.Millisecond)
+	maxConsecutiveErrors := e.ScenarioOptionInt("max-consecutive-errors", 10)
+
+	if minBacklog < 0 {
+		return fmt.Errorf("min-backlog must be non-negative")
+	}
+	if maxBacklog <= minBacklog {
+		return fmt.Errorf("max-backlog must be greater than min-backlog")
+	}
+	if frequency <= 0 {
+		return fmt.Errorf("frequency must be greater than 0")
+	}
+
+	// Activity config
+	var sleepActivityConfig *loadgen.SleepActivityConfig
+	if sleepActivitiesStr, ok := e.ScenarioOptions[SleepActivityJsonFlag]; ok {
+		var err error
+		sleepActivityConfig, err = loadgen.ParseAndValidateSleepActivityConfig(sleepActivitiesStr)
+		if err != nil {
+			return fmt.Errorf("failed to parse %s: %w", SleepActivityJsonFlag, err)
+		}
+	}
+	if sleepActivityConfig == nil {
+		sleepActivityConfig = &loadgen.SleepActivityConfig{}
+	}
+	if len(sleepActivityConfig.Groups) == 0 {
+		sleepActivityConfig.Groups = map[string]loadgen.SleepActivityGroupConfig{"default": {}}
+	}
+	for name, group := range sleepActivityConfig.Groups {
+		fixedDist := loadgen.NewFixedDistribution(sleepDuration)
+		group.SleepDuration = &fixedDist
+		sleepActivityConfig.Groups[name] = group
+	}
+
+	var consecutiveErrCount int
+	errCh := make(chan error, 10000)
+	ticker := time.NewTicker(controlInterval)
+	defer ticker.Stop()
+	var startWG sync.WaitGroup
+	iter := 1
+
+	e.Logger.Infof("Starting ebb and flow scenario: min_backlog=%d, max_backlog=%d, frequency=%v, duration=%v",
+		minBacklog, maxBacklog, frequency, e.Configuration.Duration)
+
+	isDraining := false // true = draining mode, false = growing mode
+	cycleStartTime := e.startTime
+
+	for elapsed := time.Duration(0); elapsed < e.Configuration.Duration; elapsed = time.Since(e.startTime) {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case err := <-errCh:
+			if err != nil {
+				e.Logger.Errorf("Failed to spawn workflow: %v", err)
+				consecutiveErrCount++
+				if consecutiveErrCount >= maxConsecutiveErrors {
+					return fmt.Errorf("got %v consecutive errors, most recent: %w", maxConsecutiveErrors, err)
+				}
+			} else {
+				consecutiveErrCount = 0
+			}
+		case <-ticker.C:
+			generated := e.generatedCount.Load()
+			processed := e.processedCount.Load()
+			backlog := generated - processed
```

Comment:
any concerns about the discrepancy between how this is computed and the "actual" backlog? I don't think so, and the actual number is too hard to get anyway, so nothing to do